dango/auth/src/lib.rs

```diff
@@ -66,6 +61,40 @@ pub fn authenticate_tx(
         metadata.username,
     );
 
+    verify_nonce_and_signature(ctx, tx, Some(factory), Some(metadata))
+}
+
+/// Ensure the nonce is acceptible and the signature is authentic.
+///
+/// Compared to [`authenticate_tx`](crate::authenticate_tx), this function skips
+/// the part of verifying the username.
+///
+/// This is intended for the multi-signature (Safe) account, where we ensure the
+/// username is associated with the Safe _at the time a proposal was created_,
+/// instead of _now_.
+pub fn verify_nonce_and_signature(
+    ctx: AuthCtx,
+    tx: Tx,
+    maybe_factory: Option<Addr>,
+    maybe_metadata: Option<Metadata>,
+) -> anyhow::Result<()> {
```

Comment:
My solution to #263 is to split `authenticate_tx` into two functions.

Spot and Margin accounts use `authenticate_tx`. This function:

1. ensures signer username is associated with the sender account
2. ensures nonce is correct
3. ensures signature is valid

Then a new function `verify_nonce_and_signature` is used by Safe account. This function skips (1) and only does (2) and (3).